### ISSUE TITLE
Same name for interface and class, which should be not allowed

### ISSUE THREAD

**Issue by ReyAyanami** [Microsoft/TypeScript#11141]
**TypeScript Version:**  2.0.3

**Code**

```ts
interface Person {
  say(): void;
}

class Person {

}

class Country {
  president: Person;
}

const country = new Country();
country.president = new Person;

country.president.say();


```

There should be warning or error that same name was used for interface and class.
Now it compiles without any error, but fails on runtime with: no such method.
Sure, i just can add ```I``` prefix for all interfaces, but will be great to catch this on compilation.

**Comment by aleksey-bykov** [comment]
it's a very interesting situation, what happens is the interface Person is progressively defined in 2 places, that is it has 2 definitions (one from class, one from interface) that are merged, so it's kind of a feature of TypeScript (very useful at  times), not sure if it needs to be fixed, however i see your frustration

**Comment by RyanCavanaugh** [comment]
This is intentional -- it's how you declare additional methods on a class. It's necessary to be able to do this because some libraries (not naming names) simply mutate existing class prototypes to expose their functionality.
